services/copilot.ts

```diff
@@ -0,0 +1,28 @@
+import { ClientToken, ClientTokenSchema, IUToken, IUTokenSchema, Token, WorkspaceResponse } from '@/types/common';
+import { CopilotAPI } from '@/utils/copilotApiUtils';
+import { z } from 'zod';
+
+interface Tokenable {
+  token: string;
+}
+
+// Fetch workspace from API
+export async function getWorkspaceInfo({ token }: Tokenable): Promise<WorkspaceResponse> {
```

Comment:
We have been following convention of using classes, so lets use class if possible.
Also we can instantiate the CopilotAPI class in the constructor

```
class CopilotService() {
  private copilot: CopilotAPI;

  constructor({ token }: Tokenable) {
    this.copilot = new CopilotAPI(z.string().parse(token));
  }

  getTokenPayload (): Promise<Token> {
    return await this.copilot.getTokenPayload();
  };
}

const copilotService = new CopilotService(token);
copilotService.getTokenPayload();
```

Reply:
The reason for implementing this despite the extra boilerplate was because I saw something similar in existing while onboarding from autoresponder to this project. 

Will avoid this in the future, thanks for pointing it out :stuck_out_tongue: